hbase_storage/src/main/java/com/splicemachine/derby/hbase/SpliceMultiRowRangeFilter.java

```diff
@@ -0,0 +1,273 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splicemachine.derby.hbase;
+
+import com.splicemachine.db.iapi.error.StandardException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.PrivateCellUtil;
+import org.apache.hadoop.hbase.exceptions.DeserializationException;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.filter.MultiRowRangeFilter;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.FilterProtos;
+import org.apache.hbase.thirdparty.com.google.protobuf.InvalidProtocolBufferException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.splicemachine.db.shared.common.reference.SQLState.LANG_INTERNAL_ERROR;
+
+@SuppressFBWarnings(value="HE_EQUALS_NO_HASHCODE", justification="Intentional")
+public class SpliceMultiRowRangeFilter extends MultiRowRangeFilter {
+
+    private final RangeIteration ranges;
+    private RowRange range;
+    private int index;
+    private boolean done = false;
+    private ReturnCode currentReturnCode;
+    private static final int ROW_BEFORE_FIRST_RANGE = -1;
+
+  /**
+   * @param list A list of <code>RowRange</code>
+   */
+    public SpliceMultiRowRangeFilter(List<RowRange> list) {
+        super(list);
+        this.ranges = new RangeIteration(super.getRowRanges());
+    }
+
+    @Override
+    public void reset() throws IOException {
+        done = false;
+        ranges.setFoundFirstRange(false);
+        currentReturnCode = ReturnCode.INCLUDE;
+    }
+
+    @Override
+    public boolean filterAllRemaining() {
+        return false;
+    }
+
+    @Override
+    public ReturnCode filterCell(final Cell ignored) {
+      return currentReturnCode;
+    }
+
+    @Override
+    public Cell getNextCellHint(Cell currentKV) {
+        // skip to the next range's start row
+        // #getComparisonData lets us avoid the `if (reversed)` branch
+        byte[] comparisonData = range.getComparisonData();
+        return PrivateCellUtil.createFirstOnRow(comparisonData, 0, (short) comparisonData.length);
+    }
+
+    @Override
+    public boolean filterRowKey(Cell firstRowCell) {
+        if (filterAllRemaining()) return true;
```

Comment:
this is superfluous since `filterAllRemaining` always returns false.

Reply:
True, but if the implementation is modified in the future, this would be required for peak performance.  It's good to leave this in so it's not something that has to be remembered or rediscovered.